### PR TITLE
xenlib: add sphinx/doxygen docs and build

### PR DESCRIPTION
**Pull request by GrygiriiS**
This PR adds initial version of xenlib documentation which is built
using sphinx and doxygen.

to build docs run below commands:
```
 cd docs
 mkdir build
 cd build
 cmake ../
 make [Doxygen|Sphinx]
```

The HTML docs can be found in `build/sphinx/index.html`.

The patch is based on [1] and [2], and most of doxygen and sphinx settings
just taken form Zephyr docs.

[1] https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/
[2] https://medium.com/practical-coding/c-documentation-with-doxygen-cmake-sphinx-breathe-for-those-of-use-who-are-totally-lost-part-2-21f4fb1abd9f

### CODE REVIEW

File: docs/Doxyfile.in [hunk 1]
```diff
@@ -0,0 +1,2624 @@
...
+
+DOXYFILE_ENCODING      = UTF-8
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "My Project"
```
Comment (lorc) [inline docs/Doxyfile.in:35]: Should you change this at least?

File: xenstore-srv/include/xss.h [hunk 2]
```diff
@@ -4,18 +4,34 @@
  * Copyright (c) 2023 EPAM Systems
  *
  */
+/**
+ * @file xss.h
+ */
+#ifndef XENLIB_XSS_H
+#define XENLIB_XSS_H
 
-#pragma once
```
Comment (lorc) [inline xenstore-srv/include/xss.h:8]: We had this in the our codebase? :( 

This says a lot about our review process :(  

Anyways, I'd move this change to a separate commit, as it have nothing to do with the docs.

**Review by lorc — COMMENTED** [review]
Looks great. How do you think, is it possible to integrate your changes with readthedocs.io? 

I am already hosting mouling docs here, and it would be cool to host xenlib docs as well.

**Comment by GrygiriiS** [comment]
> Looks great. How do you think, is it possible to integrate your changes with readthedocs.io?
> 
> I am already hosting mouling docs here, and it would be cool to host xenlib docs as well.

Don't know :( I really not an expert in this :) Just quickly did it on knees

File: docs/Doxyfile.in (hunk 1 above)
Comment (GrygiriiS) [reply docs/Doxyfile.in:35]: Yep. Probably not only here. Project name = "zephyr-xenlib"?

Comment (lorc) [reply docs/Doxyfile.in:35]: Yep

File: xenstore-srv/include/xss.h (hunk 2 above)
Comment (GrygiriiS) [reply xenstore-srv/include/xss.h:8]: that will go away with my misc PR

**Comment by lorc** [comment]
> Don't know :( I really not an expert in this :) Just quickly did it on knees

Okay, I'll handle that part. Could you please update the PR so we can continue with the review?

File: docs/Doxyfile.in (hunk 1 above)
Comment (GrygiriiS) [reply docs/Doxyfile.in]: done

File: xenstore-srv/include/xss.h (hunk 2 above)
Comment (GrygiriiS) [reply xenstore-srv/include/xss.h]: fixed

**Comment by GrygiriiS** [comment]
FYI. checkpatch failed due to web-links in commit message

**Review by lorc — APPROVED** [review]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Review by firscity — APPROVED** [review]
I'm not a big expert in Doxygen configuration, but I believe that you've tested it.

The only comment for first commit - AFAIK, all source and CMake should contain SPDX identifier on top lines. Please check it if is correct.

However, you can have my:
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`
